Gems/ROS2/Code/Source/RobotImporter/SDFormat/ROS2SDFormatHooksUtils.cpp

```diff
@@ -98,4 +99,167 @@ namespace ROS2::SDFormat
                 "CreatePrefabFromUrdfOrSdf", "Setting transform failed: %s does not have transform interface\n", sdfSensor.Name().c_str());
         }
     }
+
+    namespace HooksUtils::PluginParser
+    {
+        AZStd::string LastOnPath(AZStd::string path)
+        {
+            if (path.empty())
+            {
+                AZ_Warning("PluginParser", false, "Encountered empty parameter value while parsing URDF/SDF plugin.");
+                return "";
+            }
+            else if (path.contains('/'))
+            {
+                int startPos = path.find_last_of('/') + 1;
+                if (startPos >= path.size())
+                {
+                    AZ_Warning("PluginParser", false, "Encountered empty parameter value while parsing URDF/SDF plugin.");
+                    return "";
+                }
+                path = path.substr(startPos, path.size() - startPos);
+            }
+            return path;
+        }
+
+        // Inserts name (key) and value (val) of given parameter to map.
+        void ParseRegularContent(const sdf::Element& content, HooksUtils::PluginParams& remappings)
+        {
+            std::string contentName = content.GetName();
+            std::string contentValue = content.GetValue()->GetAsString();
+            if (contentName.empty() || contentValue.empty())
+            {
+                AZ_Warning("PluginParser", false, "Encountered empty parameter value while parsing URDF/SDF plugin.");
+                return;
+            }
+            AZStd::string key(contentName.c_str(), contentName.size());
+            AZStd::string val(contentValue.c_str(), contentValue.size());
+            remappings[key] = val;
+        }
+
+        // Parses parameters present in <ros> element, inserting them to map.
+        void ParseRos2Content(const sdf::Element& rosContent, HooksUtils::PluginParams& remappings)
```

Comment:
From what I understand, those functions only validate namespace and topic names using regex, not taking into account current state of namespaces (so they don't validate if the namespace to be remapped even exists). This functionality seems to come in handy when it comes to topic names, but problem becomes more complex when it comes to the namespace remaps. Some of the allowed remaps in SDFormat don't fit into the requirements of ROS (eg. `~/out:=foo`). Parsing those inaccuracies before checking them with `ValidateNamespaceField()` may cause even more chaos, especially since because of inconsistency between o3de and gazebo default namespace names, preffixes of the namespace names are not even considered in remapping imported robot's namespaces. I will try to use them in the topic remap though.